apps/aens/src/aens.erl

```diff
@@ -0,0 +1,78 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2018, Aeternity Anstalt
+%%% @doc
+%%%    Resolve registered names
+%%% @end
+%%%-------------------------------------------------------------------
+
+-module(aens).
+
+%% API
+-export([resolve/2,
+         get_name_entry/1]).
+
+%%%===================================================================
+%%% Types
+%%%===================================================================
+
+-define(LABEL_SEPARATOR, <<".">>).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+-spec resolve(atom(), binary()) -> {ok, binary()} | {error, atom()}.
+resolve(Type, Binary) ->
+    case lists:reverse(binary:split(Binary, ?LABEL_SEPARATOR, [global, trim])) of
```

Comment:
I see the `trim` option makes the last `.` ignored: may you please add a comment?
```
$ erl
Erlang/OTP 20 [erts-9.2] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:10] [hipe] [kernel-poll:false]

Eshell V9.2  (abort with ^G)
1> binary:split(<<"a.test">>, <<".">>, [global, trim]).
[<<"a">>,<<"test">>]
2> binary:split(<<"a..test">>, <<".">>, [global, trim]).
[<<"a">>,<<>>,<<"test">>]
3> binary:split(<<"a.test.">>, <<".">>, [global, trim]).
[<<"a">>,<<"test">>]
4> binary:split(<<".a.test">>, <<".">>, [global, trim]).
[<<>>,<<"a">>,<<"test">>]
5> binary:split(<<"a.test.">>, <<".">>, [global]).
[<<"a">>,<<"test">>,<<>>]
6> binary:split(<<"a.test. ">>, <<".">>, [global, trim]).
[<<"a">>,<<"test">>,<<" ">>]
...
8> binary:split(<<"">>, <<".">>, [global, trim]).
[]
9> binary:split(<<".">>, <<".">>, [global, trim]).
[]
10> binary:split(<<".a">>, <<".">>, [global, trim]).
[<<>>,<<"a">>]
11> binary:split(<<"a.">>, <<".">>, [global, trim]).
[<<"a">>]
```

Reply:
yes, this is to avoid potential user's DNS habits (where the last dot is correct)